src/services/drpcEvents/index.ts

```diff
@@ -241,4 +249,76 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleParentQuery(
+    id: UUID,
+    childQuery: QueryRow,
+    params: SubmitQueryResponseRPCParams,
+    rpcResponse?: DrpcResponse
+  ) {
+    try {
+      const [parentQuery]: QueryRow[] = await this.db.get('query', { id })
+      if (!parentQuery || parentQuery.query_response !== null) {
+        throw new Error('parent query already has a response or not found')
+      }
+
+      const [connection] = await this.db.get('connection', { id: parentQuery.connection_id })
+      if (!connection.agent_connection_id) throw new Error('missing agent_id')
+
+      const allChilds = await this.db.get('query', { parent_id: id })
+      if (!allChilds.every(({ status }) => status === 'resolved')) {
+        this.logger.info('not all childs have responded. returning.')
+        return
+      }
+      // adding childs and parent emissions
+      const total: number = parseInt(childQuery.details.emissions) + parseInt(params.emissions || '0')
```

Comment:
This is only summing the current child. As you've already fetched `allChilds` you should be able to calculate from that

Reply:
indeed, thanks for flag up, since I was testing with single forward did not notice, it meant to have TODO prefix